### PR TITLE
[BC] Multiprocessing fix and change to avoid Issue #391

### PR DESCRIPTION
**Pull request by tvmarino**
The main from generate_bc_trajectories needs to be in a separate file otherwise cloudpickle fails. To avoid Issue #391 we also parse the gin config for ```mlgo_task_type``` in each ```ModuleExplorer``` sub-process.

### CODE REVIEW

File: compiler_opt/rl/generate_bc_trajectories.py [hunk 1]
```diff
@@ -742,7 +741,7 @@ def __init__(
       clang_path: str = gin.REQUIRED,
       mlgo_task_type: Type[env.MLGOTask] = gin.REQUIRED,
       policy_paths: List[Optional[str]] = [],
-      exploration_frac: float = gin.REQUIRED,
+      exploration_frac: float = 1.0,
```
Comment (mtrofin) [inline compiler_opt/rl/generate_bc_trajectories.py:744]: wait why?

File: compiler_opt/rl/generate_bc_trajectories.py [hunk 2]
```diff
@@ -869,7 +868,7 @@ def select_best_exploration(
 def gen_trajectories(
     #  pylint: disable=dangerous-default-value
     data_path: str = gin.REQUIRED,
-    delete_flags: Tuple[str, ...] = gin.REQUIRED,
+    delete_flags: Tuple[str, ...] = (),
```
Comment (mtrofin) [inline compiler_opt/rl/generate_bc_trajectories.py:871]: why?

**Review by mtrofin — APPROVED** [review]
lgtm, curious though why default values had to change?

File: compiler_opt/rl/generate_bc_trajectories.py (hunk 2 above)
Comment (tvmarino) [reply compiler_opt/rl/generate_bc_trajectories.py:871]: For both default parameter changes -- I realized that these should not be gin.REQUIRED and they have standard default values. For exploration_frac the normal behavior is to set it to 1.0 and not constrain the number of exploration steps, for delete_flags the default behavior should be not to delete any flags.

File: compiler_opt/rl/generate_bc_trajectories.py [hunk 3]
```diff
@@ -341,6 +338,8 @@ def __init__(
           'reward_key not specified in ModuleExplorer initialization.')
     self._reward_key = reward_key
     kwargs.pop('reward_key', None)
+    gin_config_str = kwargs.pop('gin_config_str', None)
+    gin.parse_config(gin_config_str)
```
Comment (mtrofin) [inline compiler_opt/rl/generate_bc_trajectories.py:342]: add a `gin.clear_config()` here. In test, for example, previous tests that do a `gin.bind_parameter` can mess things up.

separate issue: what we need to do with `gin.bind_parameter` in tests (in a maintainable way)

**Pull request merged**